src/Core/Material/BlinnPhongMaterialModel.hpp

```diff
@@ -45,8 +54,12 @@ class RA_CORE_API BlinnPhongMaterialModel : public MaterialModel
     bool m_hasTexShininess { false };
     bool m_hasTexNormal { false };
     bool m_hasTexOpacity { false };
+
+    static std::minstd_rand m_randomEngine;
```

Comment:
Why having a static generator ? I prefer that the user gives its own generator (from simple linear congruential engine, as here, to less cyclic mersenne twister or, better, low discrepancy sequence generator) as a parameter of each sampling method. This parameter could be omitted, in which case a default mersenne twister engine should be used.

Reply:
That was David's suggestion when he saw that the generator was first declared in the function body. But I can make an optional parameter instead.